src/main/java/uk/ac/ebi/intact/psi/mi/xmlmaker/file/processing/FileFormater.java

```
@@ -86,235 +107,202 @@ public void selectFileFormater(String filePath,
-            String prey = FileUtils.getCellValueAsString(row.getCell(preyColumnIndex));
+            T row = iterator.next();
+            rowIndex++;
+
+            String bait = getBait.apply(row);
+            String prey = getPrey.apply(row);
+
+            String baitName = getBaitName.apply(row);
+            String preyName = getPreyName.apply(row);
 
-            String baitName = baitNameColumnIndex == -1 ? "" : FileUtils.getCellValueAsString(row.getCell(baitNameColumnIndex));
-            String preyName = preyNameColumnIndex == -1 ? "" : FileUtils.getCellValueAsString(row.getCell(preyNameColumnIndex));
+            if (bait.isEmpty() || prey.isEmpty()) {
+                continue;
+            }
 
             if (binary) {
                 interactionNumber++;
-                addNewParticipant(String.valueOf(interactionNumber), bait, baitName, "bait");
-                addNewParticipant(String.valueOf(interactionNumber), prey, preyName, "prey");
+                addNewParticipant(String.valueOf(interactionNumber), bait, baitName, "bait", rowIndex);
+                addNewParticipant(String.valueOf(interactionNumber), prey, preyName, "prey", rowIndex);
             } else {
                 if (lastBait == null || !lastBait.equals(bait)) {
                     interactionNumber++;
                     lastBait = bait;
-                    addNewParticipant(String.valueOf(interactionNumber), bait, baitName, "bait");
+                    addNewParticipant(String.valueOf(interactionNumber), bait, baitName, "bait", rowIndex);
                 }
-                addNewParticipant(String.valueOf(interactionNumber), prey, preyName, "prey");
+                addNewParticipant(String.valueOf(interactionNumber), prey, preyName, "prey", rowIndex);
             }
         }
-        iterator.remove();
-    }
 
+        try {
+            iterator.remove();
+        } catch (UnsupportedOperationException ignored) {
```

Why are we ignoring this exception?

---

I think I had a warning at some points but I removed it and it seems to not be here anymore